drivers/i2c/i2c_smartbond.c

```diff
@@ -515,7 +572,78 @@ static int i2c_smartbond_init(const struct device *dev)
 				       I2C_MODE_CONTROLLER | i2c_map_dt_bitrate(config->bitrate));
 }
 
+#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_PM_DEVICE_RUNTIME)
+static int i2c_smartbond_suspend(const struct device *dev)
+{
+	int ret;
+	const struct i2c_smartbond_cfg *config = dev->config;
+
+	/* Make sure there is no pending transaction */
+	i2c_smartbond_wait_until_inactive(dev);
```

Comment:
Do we still need this?

Reply:
No, at this point there should be no ongoing I2C activity. In fact, I totally removed the introduction of that API along with its subroutines. Since that changes should not reflect the PM support, they should be moved to a separate PR (along with further driver updates).